feat: update rusqlite version, add manager

SantiagoPittella opened the pull request:
This PR updates the rusqlite version to 0.35. To do so, I copied the implementation of the pool manager from the `miden-node` to this repo, since `deadpool-sqlite` does not support this version.

igamigo commented on crates/rust-client/src/store/sqlite_store/db_managment/mod.rs:
```diff
@@ -0,0 +1,7 @@
+pub(crate) mod connection;
+pub(crate) mod errors;
+pub(crate) mod migrations;
+pub(crate) mod pool_manager;
+pub(crate) mod settings;
```
Do you think we need the `migrations` and `settings` mods? It feels overkill for now, but maybe you had something else in mind.

tomyrd commented on crates/rust-client/src/store/sqlite_store/db_management/migrations.rs:
The contents of this file also seem to appear in crates/rust-client/src/store/sqlite_store/db_management/mod.rs

SantiagoPittella replied:
Oh, my bad. the `migrations.rs` file was supposed to be deleted.

tomyrd commented on crates/rust-client/src/store/sqlite_store/db_management/mod.rs:
````diff
@@ -0,0 +1,185 @@
+// UTILS
+// ================================================================================================
+
+/// Auxiliary macro which substitutes `$src` token by `$dst` expression.
+macro_rules! subst {
+    ($src:tt, $dst:expr_2021) => {
+        $dst
+    };
+}
+
+/// Generates a simple insert SQL statement with parameters for the provided table name and fields.
+/// Supports optional conflict resolution (adding "| REPLACE" or "| IGNORE" at the end will generate
+/// "OR REPLACE" and "OR IGNORE", correspondingly).
+///
+/// # Usage:
+///
+/// ```ignore
+/// insert_sql!(users { id, first_name, last_name, age } | REPLACE);
+/// ```
+///
+/// which generates:
+/// ```sql
+/// INSERT OR REPLACE INTO `users` (`id`, `first_name`, `last_name`, `age`) VALUES (?, ?, ?, ?)
+/// ```
+macro_rules! insert_sql {
+    ($table:ident { $first_field:ident $(, $($field:ident),+)? $(,)? } $(| $on_conflict:expr)?) => {
+        concat!(
+            stringify!(INSERT $(OR $on_conflict)? INTO ),
+            "`",
+            stringify!($table),
+            "` (`",
+            stringify!($first_field),
+            $($(concat!("`, `", stringify!($field))),+ ,)?
+            "`) VALUES (",
+            subst!($first_field, "?"),
+            $($(subst!($field, ", ?")),+ ,)?
+            ")"
+        )
+    };
+}
````
Should this be moved to `utils.rs`? Plus, I think we could use the `insert_sql` macro in our sqlite store code.

SantiagoPittella replied:
>Should this be moved to utils.rs?
I think that is small enough to keep it in this mod, wdyt?

> I think we could use the insert_sql macro in our sqlite store code
Will see where to replace it

tomyrd reviewed (commented):
The CI failed tests should be fixed once with a merge from the `next` changes.

igamigo reviewed (commented):
Overall looks great, I like that we removed `deadpool-sqlite` especially because the code that replaces it is not a lot.
However, I think there are some things that we could still remove and simplify. The migration related code I think is a bit complex for what we need right now, so I wonder if we should leave it out (we used a crate that we eventually removed at some point). On the other hand, it is something we will probably eventually need. 

Also, if we keep some of the code (like the `insert_sql` macro), we should make an issue to use it more consistently throughout the codebase.

SantiagoPittella commented:
I removed the `Connection` and `Transaction` structs, it was not adding any value. Replaced them with the native `rusqlite` ones. Also removed the `Settings` struct in favor of a couple extra utils function. I keep the `Migrations` struct though, since I found it useful and don't think it's so much code that it would be heavy to maintain.

Let me know  your opinions @igamigo @tomyrd

igamigo reviewed (approved):
LGTM!
